test/new-e2e/tests/installer/upgrade_scenario_test.go

```diff
@@ -501,6 +560,117 @@ func (s *upgradeScenarioSuite) assertSuccessfulAgentStopExperiment(timestamp hos
+	output, err := s.startConfigExperiment(pkg, version)
+	require.NoError(s.T(), err, "Failed to start config experiment: %s\ndatadog-installer journalctl:\n%s\ndatadog-installer-exp journalctl:\n%s",
+		s.Env().RemoteHost.MustExecute("cat /tmp/start_config_experiment.log"),
+		s.Env().RemoteHost.MustExecute("sudo journalctl -xeu datadog-installer --no-pager"),
+		s.Env().RemoteHost.MustExecute("sudo journalctl -xeu datadog-installer-exp --no-pager"),
+	)
+	return output
+}
+
+func (s *upgradeScenarioSuite) promoteConfigExperiment(pkg packageName) (string, error) {
+	cmd := fmt.Sprintf("sudo datadog-installer daemon promote-config-experiment %s > /tmp/promote_config_experiment.log 2>&1", pkg)
+	s.T().Logf("Running promote command: %s", cmd)
+	return s.Env().RemoteHost.Execute(cmd)
+}
+
+func (s *upgradeScenarioSuite) mustPromoteConfigExperiment(pkg packageName) string {
+	output, err := s.promoteConfigExperiment(pkg)
+	require.NoError(s.T(), err, "Failed to promote config experiment: %s\ndatadog-installer journalctl:\n%s\ndatadog-installer-exp journalctl:\n%s",
+		s.Env().RemoteHost.MustExecute("cat /tmp/promote_config_experiment.log"),
+		s.Env().RemoteHost.MustExecute("sudo journalctl -xeu datadog-installer --no-pager"),
+		s.Env().RemoteHost.MustExecute("sudo journalctl -xeu datadog-installer-exp --no-pager"),
+	)
+	return output
+}
+
+func (s *upgradeScenarioSuite) stopConfigExperiment(pkg packageName) (string, error) {
+	cmd := fmt.Sprintf("sudo datadog-installer daemon stop-config-experiment %s > /tmp/stop_config_experiment.log 2>&1", pkg)
+	s.T().Logf("Running stop command: %s", cmd)
+	return s.Env().RemoteHost.Execute(cmd)
+}
+
+func (s *upgradeScenarioSuite) mustStopConfigExperiment(pkg packageName) string {
+	output, err := s.stopConfigExperiment(pkg)
+	require.NoError(s.T(), err, "Failed to stop experiment: %s\ndatadog-installer journalctl:\n%s\ndatadog-installer-exp journalctl:\n%s",
+		s.Env().RemoteHost.MustExecute("cat /tmp/stop_config_experiment.log"),
+		s.Env().RemoteHost.MustExecute("sudo journalctl -xeu datadog-installer --no-pager"),
+		s.Env().RemoteHost.MustExecute("sudo journalctl -xeu datadog-installer-exp --no-pager"),
+	)
+	return output
+}
+
+func (s *upgradeScenarioSuite) assertSuccessfulConfigStartExperiment(timestamp host.JournaldTimestamp, hash string) {
+	s.host.WaitForUnitActivating(agentUnitXP)
+	s.host.WaitForFileExists(false, "/opt/datadog-packages/datadog-agent/experiment/run/agent.pid")
+
+	// Assert experiment is running
+	s.host.AssertSystemdEvents(timestamp, host.SystemdEvents().
+		Unordered(host.SystemdEvents().
+			Stopped(agentUnit).
+			Stopped(traceUnit).
+			Stopped(processUnit),
+		).
+		Unordered(host.SystemdEvents().
+			Starting(agentUnitXP).
+			Started(traceUnitXP).
+			Started(processUnitXP),
+		),
+	)
+
+	state := s.host.State()
+	state.AssertSymlinkExists("/etc/datadog-packages/datadog-agent/experiment", fmt.Sprintf("/etc/datadog-packages/datadog-agent/%s", hash), "root", "root")
+}
+
```

Comment:
you could add a datadog.yaml config with no api_key to verify that the experiment revert on crash